src/helpers/notification_helpers.ts

```diff
@@ -1,45 +1,143 @@
+    }
+
+    await add__new_sms_to_db(number, text, "multisend", msgId);
+    logger.log("send_local_sms. message sent successfully", { number, text, response: response.data });
+};
+
+const send_international_sms: SmsFunction = async (number, text) => {
+    const {
+        sms_provider: { twilio },
+    } = cache_manager.getObjectData("nx-settings");
+    const twilioClient = new Twilio(twilio.account_sid, twilio.token);
+    const message = await twilioClient.messages.create({
+        messagingServiceSid: twilio.messaging_service_sid,
+        body: text,
+        to: number,
+    });
+    if (message.errorMessage) {
+        throw `twilioClient.messages.create failed: ${message.errorMessage} `;
+    }
+    await add__new_sms_to_db(number, text, "twilio", message.sid);
+    logger.log("send_international_sms. message sent successfully", { number, text, response: message });
+};
+
+const login_to_monogoto = async () => {
     try {
-        const { sms_provider } = cache_manager.getObjectData("nx-settings");
-        let data = new FormData();
-        data.append("user", sms_provider.user);
-        data.append("password", sms_provider.password);
-        data.append("from", sms_provider.from);
-        data.append("recipient", phone_number);
-        data.append("message", text);
-        let config = {
+        const {
+            sms_provider: { monogoto },
+        } = cache_manager.getObjectData("nx-settings");
+        const data = { UserName: monogoto.UserName, Password: monogoto.Password };
+
+        const response = await axios({
             method: "post",
-            maxBodyLength: Infinity,
-            url: "https://api.multisend.co.il/v2/sendsms",
-            data: data,
+            url: `https://console.monogoto.io/Auth`,
+            data,
+        });
+        return response.data;
+    } catch (error) {
+        throw `login_to_monogoto failed: ` + error;
+    }
+};
+
+const send_iccid_sms: SmsFunction = async (number, text) => {
+    const {
+        sms_provider: { monogoto },
+    } = cache_manager.getObjectData("nx-settings");
+    const monogoto_auth = await login_to_monogoto();
+    const data = { Message: text, From: monogoto.from };
+    const response = await axios({
+        method: "post",
+        url: `https://console.monogoto.io/thing/ThingId_ICCID_${number}/sms`,
+        data: data,
+        headers: {
+            Authorization: `Bearer ${monogoto_auth.token}`,
+            apikey: monogoto_auth.CustomerId,
+        },
+    });
+    if (response.status !== 200) {
+        throw `request to monogoto status: ${response.status}`;
+    }
+    await add__new_sms_to_db(number, text, "monogoto", response.data);
+    logger.log("send_iccid_sms. message sent successfully", { number, text, response: response.data });
+};
+
+export const send_sms = async (number: string, text: string, entity_for_audit: string) => {
+    try {
+        let service = "multisend";
```

Comment:
make following code better. this line does not look good. may be make your send() function return a service used inside. Also, SmsFunction does returns nothing, hence return void looks confusing. I understand it works, but better return "service"

Reply:
done